executor/join.go

```diff
@@ -74,11 +78,24 @@ type HashJoinExec struct {
 
 	memTracker *memory.Tracker // track memory usage.
 
-	// radixBits indicates the bit number using for radix partitioning. Inner
-	// relation will be split to 2^radixBits sub-relations before building the
-	// hash tables. If the complete inner relation can be hold in L2Cache in
-	// which case radixBits will be 0, we can skip the partition phase.
-	radixBits int
+	// radixBits indicates the bits using for radix partitioning. Inner relation
+	// will be split to 2^radixBitsNumber sub-relations before building the hash
+	// tables. If the complete inner relation can be hold in L2Cache in which
+	// case radixBits will be 1, we can skip the partition phase.
+	radixBits       uint32
+	innerParts      []partition
+	innerRowPrts    [][]*ptr4Partition
```

Comment:
`ptr4Partition` only takes 24 bytes, this slice is heavily used in the partition phase, maybe it's better to change the definition from `[][]*ptr4Partition` to `[][]ptr4Partition` to improve the cache locality.